### ISSUE TITLE
False positive on clojure.test.check.properties/for-all

### ISSUE THREAD

**Issue by carrete** [borkdude/clj-kondo#598]
**version**

2019.11.07

**platform**

Run on the JVM as shown in this commit: https://gitlab.com/tvaughan/kibit-runner/commit/8399d7af94cf07ca3cadc599a831092c22dfb749

**editor**

N/A

**problem**

clj-kondo reports:

    test/kibit_runner/utils_test.clj:14:18: error: unresolved symbol paths

on:

```clojure
(prop/for-all [paths (gen/not-empty (gen/vector gen/string-alphanumeric))]
                (= (map #(.getName %) (utils/parse-paths (string/join "," paths))) paths))
```

**repro**

https://gitlab.com/tvaughan/kibit-runner

Failed pipeline job: https://gitlab.com/tvaughan/kibit-runner/-/jobs/345933624

**expected behavior**

clj-kondo should not report that paths is an unresolved symbol. I suspect the problem is that clj-kondo interprets `[paths (gen/not-empty (gen/vector gen/string-alphanumeric))]` as a vector.

**Comment by borkdude** [comment]
@carrete This false positive can be prevented using configuration:

```
$ cat /tmp/forall.clj
(require '[clojure.test.check.properties :as prop])

(prop/for-all [paths (gen/not-empty (gen/vector gen/string-alphanumeric))]
              (= (map #(.getName %) (utils/parse-paths (string/join "," paths))) paths))
```

```
$ clj-kondo --lint /tmp/forall.clj --config '{:lint-as {clojure.test.check.properties/for-all clojure.core/let}}'
linting took 12ms, errors: 0, warnings: 0
```

Configuration stored in `.clj-kondo/config.edn` is automatically picked up by clj-kondo.

**Comment by carrete** [comment]
> This false positive can be prevented using configuration:

Awesome! Thanks!
